omnigibson/prims/prim_base.py

```diff
@@ -128,6 +128,9 @@ def remove(self):
         # Also clear the name so we can reuse this later
         self.remove_names(include_all_owned=True)
 
+        # Make the simulator refresh its handles
+        og.sim.update_handles()
```

Comment:
shouldn't this occur from the sim side of things? (e.g.: within the Simulator class)?

Reply:
Ideally, yes. Unfortunately right now the removal routines don't remove objects through the og.sim.remove_object interface but instead just by calling .remove() on the individual object. Agreed we should fix this. I filed #350 to track this, but out of scope here.